File: app/src/main/java/com/example/david/lists/di/view/MainActivityComponent.java
package com.example.david.lists.di.view;

import com.example.david.lists.ui.MainActivity;

import dagger.BindsInstance;
import dagger.Component;

@MainActivityScope
@Component(modules = {MainActivityModule.class})
public interface MainActivityComponent {
    void inject(MainActivity mainActivity);

    @Component.Builder
    interface Builder {
        MainActivityComponent build();

        @BindsInstance
        Builder mainActivity(MainActivity mainActivity);
    }
}

File: app/src/main/java/com/precopia/david/lists/view/common/buildlogic/MainActivityComponent.kt
package com.precopia.david.lists.view.common.buildlogic

import android.app.Application
import com.precopia.david.lists.common.buildlogic.ViewCommonModule
import com.precopia.david.lists.common.buildlogic.ViewScope
import com.precopia.david.lists.view.common.MainActivity
import dagger.BindsInstance
import dagger.Component

@ViewScope
@Component(modules = [
    ViewCommonModule::class
])
interface MainActivityComponent {
    fun inject(activity: MainActivity)

    @Component.Builder
    interface Builder {
        fun build(): MainActivityComponent

        @BindsInstance
        fun application(application: Application): Builder
    }
}